bindsnet/datasets/__init__.py

```diff
@@ -1,945 +1,9 @@
-            if self.download:
-                print('Downloading CIFAR-100 data.\n')
-                self._download(CIFAR100.url, CIFAR100.data_archive)
-                images, labels = self.process_data(CIFAR100.test_files)
-
-                # Serialize image data on disk for next time.
-                torch.save((images, labels), open(path, 'wb'))
-            else:
-                msg = 'Dataset not found on disk; specify \'download=True\' to allow downloads.'
-                raise FileNotFoundError(msg)
-        else:
-            if not os.path.isdir(path):
-                # Process image and label data if pickled file doesn't exist.
-                images, labels = self.process_data(CIFAR100.test_files)
-
-                # Serialize image data on disk for next time.
-                torch.save((images, labels), open(path, 'wb'))
-            else:
-                # Load image data from disk if it has already been processed.
-                print('Loading test images from serialized object file.\n')
-                images, labels = torch.load(open(path, 'rb'))
-
-        if self.shuffle:
-            perm = np.random.permutation(np.arange(labels.shape[0]))
-            images, labels = images[perm], labels[perm]
-
-        return torch.Tensor(images), torch.Tensor(labels)
-
-    def _download(self, url: str, filename: str) -> None:
-        # language=rst
-        """
-        Downloads and unzips all CIFAR-100 data.
-
-        :param url: The URL of the data archive to be downloaded.
-        :param file_name: The name of the file to store extract the archive to.
-        """
-        urlretrieve(url, os.path.join(self.path, filename))
-        tar = tarfile.open(os.path.join(self.path, filename), 'r:gz')
-        tar.extractall(path=self.path)
-        tar.close()
-
-    def process_data(self, file_names: List[str]) -> Tuple[torch.Tensor, torch.Tensor]:
-        # language=rst
-        """
-        Opens files of CIFAR-100 data and processes them into :code:`numpy` arrays.
-
-        :param file_names: Name of the file containing CIFAR-100 images and labels to load.
-        :return: Processed CIFAR-100 image and label data.
-        """
-        d = {'data': [], 'labels': []}
-        for filename in file_names:
-            with open(os.path.join(self.data_path, filename), 'rb') as f:
-                temp = p.load(f, encoding='bytes')
-                d['data'].append(temp[b'data'].reshape(-1, 3, 32, 32).transpose(0, 2, 3, 1))
-                d['labels'].append(temp[b'fine_labels'])
-
-        data, labels = np.concatenate(d['data']), np.concatenate(d['labels'])
-        return data.astype(np.float32), labels.astype(np.float32)
+MNIST = vd_wrapper_creator('MNIST')
+CIFAR10 = vd_wrapper_creator('CIFAR10')
+CIFAR100 = vd_wrapper_creator('CIFAR100')
+FashionMNIST = vd_wrapper_creator('FashionMNIST')
+ImageFolder = vd_wrapper_creator('ImageFolder')
 
+from .spike_encoders import *
```

Comment:
Why import everything here? Might want to do `__all__` = ['spike_encoder']`, which allows Python to find the submodule. See e.g. [here](https://github.com/Hananel-Hazan/bindsnet/blob/master/bindsnet/__init__.py).

Reply:
I can change it to that.